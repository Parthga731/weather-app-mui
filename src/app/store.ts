import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import weatherReducer from "../slices/weatherSlice";
import logger from "redux-logger";

export const store = configureStore({
  reducer: {
    weather: weatherReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
