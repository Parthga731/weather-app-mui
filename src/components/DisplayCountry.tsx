import { Box, Button, Typography } from "@mui/material";
import { AnyMap } from "immer/dist/internal";
import { useEffect } from "react";
import { useNavigate, useParams } from "react-router-dom";
import { useAppDispatch, useAppSelector } from "../app/hooks";
import { getCountryName } from "../slices/weatherSlice";
import { BoxPaper } from "./Atomic/BoxPaper";
import { TypographyName } from "./Atomic/TypographyName";

export const DisplayCountry = () => {
  const { countryInfo, isCountryDataShow } = useAppSelector(
    (state) => state.weather
  );

  const dispatch = useAppDispatch();
  const { name } = useParams();
  const naviagte = useNavigate();
  console.log(countryInfo);

  useEffect(() => {
    dispatch(getCountryName(name));
  }, [dispatch, name]);

  return (
    <>
      {isCountryDataShow &&
        countryInfo.map((country: any, id: any) => {
          return (
            <Box key={id} data-testid={`countrylist-${id}`}>
              <BoxPaper>
                <Typography variant="h4" component="h3">
                  Country Information
                </Typography>
                <TypographyName name="Country" value={country.name} />
                <TypographyName name="Capital" value={country.capital} />
                <TypographyName name="Population" value={country.population} />
                <TypographyName name="Latlng" value={String(country.latlng)} />

                <img src={`${country.flags.png}`} alt="country photos" />
                <Button
                  variant="contained"
                  onClick={() => naviagte(`/${country.name}/weather`)}
                  sx={{ m: 3 }}>
                  Check Weather
                </Button>
              </BoxPaper>
            </Box>
          );
        })}
    </>
  );
};
