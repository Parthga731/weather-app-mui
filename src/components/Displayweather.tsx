import React, { useEffect } from "react";
import { useParams } from "react-router-dom";
import { useAppDispatch, useAppSelector } from "../app/hooks";
import { getCountryWeather } from "../slices/weatherSlice";
import { TypographyName } from "./Atomic/TypographyName";
import { Typography } from "@mui/material";
import { BoxPaper } from "./Atomic/BoxPaper";

export const DisplayWeather = () => {
  const { searchText, weatherDetail, isWeatherDataShow } = useAppSelector(
    (state) => state.weather
  );
  const dispatch = useAppDispatch();
  const { name } = useParams();
  console.log(weatherDetail);

  useEffect(() => {
    dispatch(getCountryWeather(name));
  }, [dispatch, name]);

  return (
    <>
      {isWeatherDataShow && (
        <BoxPaper>
          <Typography variant="h4" component="h3">
            Weather Information {name}
          </Typography>
          <TypographyName name="Capital" value={weatherDetail.location.name} />
          <TypographyName
            name="Temperature"
            value={weatherDetail.current.temperature}
            caption="&#8451;"
          />
          <TypographyName
            name="Weather Icon"
            value={
              <img
                src={`${weatherDetail.current.weather_icons}`}
                alt="icon"
                height="20px"
              />
            }
          />
          <TypographyName
            name="Wind Speed"
            value={weatherDetail.current.wind_speed}
            caption="KMPH"
          />
          <TypographyName name="Precip" value={weatherDetail.current.precip} />
        </BoxPaper>
      )}
    </>
  );
};
