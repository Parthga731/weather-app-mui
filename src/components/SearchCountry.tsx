import React from "react";
import { TextField, Button, Box, Typography } from "@mui/material";
import { useAppDispatch, useAppSelector } from "../app/hooks";
import { searchCountryAction } from "../slices/weatherSlice";
import { useNavigate } from "react-router-dom";

export const SearchCountry = () => {
  const { searchText } = useAppSelector((state) => state.weather);
  const dispatch = useAppDispatch();
  const navigate = useNavigate();

  const handleSubmit = (e: any) => {
    e.preventDefault();
    navigate(`./${searchText}`);
  };

  return (
    <>
      <form onSubmit={handleSubmit}>
        <Box
          sx={{ width: "100%", height: "600px" }}
          display={"flex"}
          flexDirection={"column"}
          justifyContent={"center"}
          alignItems={"center"}>
          <Typography variant="h4" component="h3" sx={{ m: 2 }}>
            Weather App
          </Typography>
          <TextField
            size="small"
            id="standard-basic"
            value={searchText}
            onChange={(e) => dispatch(searchCountryAction(e.target.value))}
            placeholder="Enter Country"
            aria-label="cost-input"
          />
          <Button
            type="submit"
            variant="contained"
            color="primary"
            disabled={!searchText}
            sx={{ m: 2 }}>
            Submit
          </Button>
        </Box>
      </form>
    </>
  );
};
