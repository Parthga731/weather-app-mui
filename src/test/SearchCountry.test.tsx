// import React from "react";

// import { SearchCountry } from "../components/SearchCountry";
// import { render, screen, fireEvent } from "./test-utils";

// const mockedUsedNavigate = jest.fn();
// jest.mock("react-router-dom", () => ({
//   ...jest.requireActual("react-router-dom"),
//   useNavigate: () => mockedUsedNavigate,
// }));

// describe("Search country components test", () => {
//   it("title is present or not", async () => {
//     render(<SearchCountry />);
//     const sub = await screen.findByText(/weather app/i);
//     expect(sub).toBeInTheDocument();
//   });
//   test("submit button disable or not", async () => {
//     render(<SearchCountry />);
//     const sub = await screen.findByText(/submit/i);
//     expect(sub).toBeDisabled();
//   });
//   test("textfield available or not", async () => {
//     render(<SearchCountry />);
//     const sub = await screen.findByPlaceholderText(/enter country/i);
//     screen.debug(sub);
//     fireEvent.change(sub, { target: { value: "india" } });
//     expect(sub.value).toBe("india");
//   });
//   test("submit click event", async () => {
//     render(<SearchCountry />);
//     const inputElement = await screen.findByPlaceholderText(/enter country/i);
//     const buttonElement = await screen.findByRole("button", {
//       name: /submit/i,
//     });
//     fireEvent.change(inputElement, { target: { value: "india" } });
//     fireEvent.click(buttonElement);
//     expect(inputElement.value).toBe("india");
//   });
// });

import * as React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { store } from "../app/store";
import ReactDOM from "react-dom";
import { SearchCountry } from "../components/SearchCountry";

describe("details snapshot ", () => {
  let container: HTMLDivElement;

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
    ReactDOM.render(
      <BrowserRouter>
        <Provider store={store}>
          <SearchCountry />
        </Provider>
      </BrowserRouter>,
      container
    );
  });

  afterEach(() => {
    document.body.removeChild(container);
    container.remove();
  });

  it("final", () => {
    expect(container).toMatchSnapshot();
  });
});
