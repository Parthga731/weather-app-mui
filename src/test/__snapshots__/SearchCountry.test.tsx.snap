// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`details snapshot  final 1`] = `
<div>
  <form>
    <div
      class="MuiBox-root css-1rr4bpd"
    >
      <h3
        class="MuiTypography-root MuiTypography-h4 css-b4jtw2-MuiTypography-root"
      >
        Weather App
      </h3>
      <div
        aria-label="cost-input"
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
            id="standard-basic"
            placeholder="Enter Country"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-nnbavb"
            >
              <span
                class="notranslate"
              >
                â€‹
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled  css-s3awaj-MuiButtonBase-root-MuiButton-root"
        disabled=""
        tabindex="-1"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
